tools/autograd/gen_python_functions.py

```diff
@@ -1349,7 +1351,28 @@ def go(f: NativeFunction) -> str:
         )
 
         if lambda_return == "void":
-            return f"""\
+            # Make in-place foreach return `self` at python-binding level.
+            # ref: https://github.com/pytorch/pytorch/pull/118622#pullrequestreview-1904804954
+            self_arg = f.func.arguments.self_arg
+            if (
+                str(f.func.name).startswith("_foreach_")
+                and f.func.kind() == SchemaKind.inplace
+                and (
+                    self_arg is not None and is_tensor_list_type(self_arg.argument.type)
```

Comment:
Would it possible to assert that if we are foreach and inplace that self_arg is None and self is tensorlist?